app/services/ui_notifications/report_finished.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module UINotifications
+  class ReportFinished < UINotifications::Base
+    include Rails.application.routes.url_helpers
+
+    def initialize(composer, job_id)
+      @composer = composer
+      @job_id = job_id
+      @subject = User.current
+    end
+
+    private
+
+    def create
+      add_notification
+    end
+
+    def update_notifications
+      blueprint.notifications.
+          where(subject: subject).
+          update_all(expired_at: blueprint.expired_at)
+    end
+
+    def add_notification
+      Notification.create!(
+        initiator: initiator,
+        audience: ::Notification::AUDIENCE_USER,
+        subject: subject,
+        message: N_(' %s is ready to download') % @composer.template,
```

Comment:
I don't think this space is intentional. Also, I think it should be clear that it's a report. How about:
```suggestion
        message: N_('Report "%s" is ready to download') % @composer.template,
```

Reply:
also `@composer.template.name` would IMHO be better for readability, who knows when someone redefines the `to_s` method on it.